packages/iris-grid/src/PartitionedGridModel.ts

```diff
@@ -43,9 +43,14 @@ export interface PartitionedGridModelProvider extends IrisGridModel {
   /** Whether a partition is required to be set on this model */
   get isPartitionRequired(): boolean;
 
+  /** Whether the table is a partition-aware source table or a partition table */
+  get isPartitionAwareSourceTable(): boolean;
+
   /** Get a keys table for the partitions */
   partitionKeysTable: () => Promise<dh.Table>;
 
+  partitionBaseTable: () => Promise<dh.Table>;
```

Comment:
Need a JS doc here. Something like:
```suggestion
  /** Get the base table for the partitions. Includes key columns and possibly other columns, such as constituent columns */
  partitionBaseTable: () => Promise<dh.Table>;
```